hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/auth/AssumedRoleCredentialProvider.java

```diff
@@ -125,37 +126,40 @@ public AssumedRoleCredentialProvider(@Nullable URI fsUri, Configuration conf)
     String policy = conf.getTrimmed(ASSUMED_ROLE_POLICY, "");
 
     LOG.debug("{}", this);
-    STSAssumeRoleSessionCredentialsProvider.Builder builder
-        = new STSAssumeRoleSessionCredentialsProvider.Builder(arn, sessionName);
-    builder.withRoleSessionDurationSeconds((int) duration);
+
+    AssumeRoleRequest.Builder requestBuilder =
+        AssumeRoleRequest.builder().roleArn(arn).roleSessionName(sessionName)
+            .durationSeconds((int) duration);
+
     if (StringUtils.isNotEmpty(policy)) {
       LOG.debug("Scope down policy {}", policy);
-      builder.withScopeDownPolicy(policy);
+      // TODO: Don't see an equivalent in V2.
+     // requestBuilder.withScopeDownPolicy(policy);
```

Comment:
Is this different from `.policy()`?

Reply:
ah good point! I guess previously this was used to attach  a policy to `STSAssumeRoleSessionCredentialsProvider`, but all config has been removed from the cred provided in favour of the assume role request, so i should just be able to attach it there with .policy()